File: lib/prompts/templates/keyPoints.ts
import type { Blueprint, GenerateRequest, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest
): string {
  const { lesson_metadata } = req
  return `Generate the key points summary for this study guide.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Core Concept: ${blueprint.core_concept}
- Learning Targets:
${blueprint.learning_targets.map((lt) => `  ${lt.number}. I can ${lt.bloom_verb} ${lt.objective}`).join('\n')}
- Sub-competencies: ${blueprint.sub_competencies.map((sc) => `${sc.label}`).join('; ')}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}

## Instructions
Write exactly 3 key point statements that:
1. Summarise the most important takeaways from this lesson
2. Correspond directly to the 3 learning targets
3. Are concise, clear statements (1 sentence each) that a student could use for review
4. Use bold-worthy language that stands out in the document
5. Are appropriate for Grade ${lesson_metadata.grade_level}

## Output Schema
{
  "key_points": [
    { "number": 1, "statement": "string" },
    { "number": 2, "statement": "string" },
    { "number": 3, "statement": "string" }
  ]
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/key_points.py
from __future__ import annotations

from app.types import Blueprint, GenerateRequest


def build_prompt(spec, blueprint: Blueprint, request: GenerateRequest) -> str:
    del spec, request

    key_points_schema = """{
    "title": "Key Points",
    "points": [
        {
            "number": 1,
            "sub_competency_id": "string",
            "sub_competency_label": "string",
            "statement": "string"
        }
    ]
}"""

    sub_competency_text = "\n".join(
        f"- {item.id}: {item.label}" for item in blueprint.sub_competencies
    )

    prompt_lines = [
        "Create the key points section for a K-12 study guide.",
        f"- Lesson title: {blueprint.title}",
        f"- Essential question: {blueprint.essential_question}",
        f"- Core concept: {blueprint.core_concept}",
        "- Sub-competencies:",
        sub_competency_text,
        "Requirements:",
        "- Write one numbered summary statement per sub-competency.",
        "- Keep statements concise, clear, and directly aligned to the listed sub-competencies.",
        "- Use the same order as the blueprint sub-competencies.",
        "- Make each statement suitable for quick review at the end of the lesson.",
        "Expected JSON schema:",
        key_points_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
